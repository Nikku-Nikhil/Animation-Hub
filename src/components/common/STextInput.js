import { StyleSheet, TextInput, View } from 'react-native'
import React from 'react'
import SText from './SText';

const STextInput = ({
    label='',
    placeholder="Tap to Write",
    value='',
    onChangeText=()=>{},
    type='ascii-capable',
    containerStyle={},
}) => {
  return (
   <View style={styles.container}>
   {label && (
    <SText text={label} size={14}/>
  )}
  <TextInput
      placeholder={placeholder}
      value={value}
      onChangeText={onChangeText}
      keyboardType={type}
      style={[containerStyle,styles.textBox]}
      placeholderTextColor={"#7F7F7F"}
    />
   </View>
  )
}

export default STextInput

const styles = StyleSheet.create({
  container:{
    marginVertical:5,
  },  
    textBox:{
        paddingVertical:13,
        paddingHorizontal:20,
        backgroundColor:"#EEEEEE",
        fontSize:16,
        color:"#000000", 
        borderRadius:8,
        marginTop:5,
        borderWidth:0.5,
    },
})