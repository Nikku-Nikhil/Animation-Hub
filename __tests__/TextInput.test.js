import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import STextInput from "../src/components/common/STextInput";

const initialProps = {
  label: "",
  placeholder: "Tap to Write",
  value: "",
  onChangeText: () => {},
  type: "ascii-capable",
  containerStyle: {},
};

describe("STextInput (common) component test cases", () => {
  //snapshot testing
  test("Renderes Correctly", () => {
    const snapshot = renderer.create(<STextInput />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test("Is label prop not present", () => {
    const title = renderer.create(<STextInput {...initialProps} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.label).toBe("");
  });

  test("Is label prop present", () => {
    const title = renderer.create(<STextInput label={"Name"} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.label).toBe("Name");
  });

  test("Is placeholder Prop is not present",()=>{
    const placeholder = renderer.create(<STextInput {...initialProps} />);
    expect(placeholder.root.findByType(STextInput).props.placeholder).toBe('Tap to Write');
  })

  test("Is placeholder Prop is present",()=>{
    const placeholder = renderer.create(<STextInput placeholder="Enter you name" />);
    expect(placeholder.root.findByType(STextInput).props.placeholder).toBe('Enter you name');
  })

  test("Is value prop not present", () => {
    const title = renderer.create(<STextInput {...initialProps} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.value).toBe("");
  });

  test("Is value prop present", () => {
    const title = renderer.create(<STextInput value={"Nikhil"} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.value).toBe("Nikhil");
  });

  test("Is type prop not present", () => {
    const title = renderer.create(<STextInput {...initialProps} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.type).toBe("ascii-capable");
  });

  test("Is type prop present", () => {
    const title = renderer.create(<STextInput type={"default"} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.type).toBe("default");
  });

  test("Handle OnChangeText Correctly", () => {
    const onChangeMockText = jest.fn();
    const tree = renderer.create(
      <STextInput onChangeText={onChangeMockText} />
    );
    tree.root.findByType(STextInput).props.onChangeText("Hello World");
    expect(onChangeMockText).toHaveBeenCalledWith("Hello World");
  });

  test("Is containerStyle prop not present", () => {
    const title = renderer.create(<STextInput {...initialProps} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.containerStyle).toStrictEqual({});
  });

  test("Is containerStyle prop present", () => {
    const title = renderer.create(<STextInput containerStyle={{borderRadius:7}} />);
    const textInput = title.root.findByType(STextInput);
    expect(textInput.props.containerStyle).toStrictEqual({borderRadius:7});
  });
});
