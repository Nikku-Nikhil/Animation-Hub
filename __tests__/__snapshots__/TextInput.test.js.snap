// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STextInput (common) component test cases Renderes Correctly 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "width": 710,
    }
  }
>
  <TextInput
    keyboardType="ascii-capable"
    onChangeText={[Function]}
    placeholder="Tap to Write"
    placeholderTextColor="#7F7F7F"
    style={
      [
        {},
        {
          "backgroundColor": "#EEEEEE",
          "borderRadius": 8,
          "borderWidth": 0.5,
          "color": "#000000",
          "fontSize": 16,
          "marginVertical": 10,
          "paddingHorizontal": 20,
          "paddingVertical": 13,
        },
      ]
    }
    value=""
  />
</View>
`;
